packages/thumbprint-react/components/Image/index.jsx

```diff
@@ -0,0 +1,234 @@
+        // following technique: https://css-tricks.com/aspect-ratio-boxes/
+        const h = 100000;
+        const w = h * aspectRatio;
+
+        containerProps.style = {
+            ...containerProps.style,
+            paddingTop: `${(h / w) * 100}%`,
+            overflow: 'hidden',
+            position: 'relative',
+        };
+
+        pictureProps.style = {
+            ...pictureProps.style,
+            position: 'absolute',
+            top: 0,
+            right: 0,
+            bottom: 0,
+            left: 0,
+            height: '100%',
+        };
+    }
+
+    if (disableLazyLoading) {
+        picture = <Picture {...pictureProps} />;
+    } else {
+        const parent = canUseDOM && scrollparent(node);
+        // If `scrollparent` doesn't find a custom scroll parent, then we just send `null` so
+        // that Intersection Observer just uses the default `window`.
+        const root =
+            parent && (parent.tagName === 'HTML' || parent.tagName === 'BODY') ? null : parent;
+
+        picture = (
+            <LazyImage
+                {...pictureProps}
+                root={root}
+                onEnter={() => {
+                    setHasImageStartedLoading(true);
+                }}
+            >
+                {lazyImageProps => <Picture {...lazyImageProps} />}
+            </LazyImage>
+        );
+    }
+
+    return (
+        <div
+            {...containerProps}
+            ref={el => {
+                setRef(el);
+
+                if (outerRef) {
+                    outerRef(el);
+                }
+            }}
+        >
+            {picture}
+        </div>
+    );
+});
+
+Image.propTypes = {
+    /**
+     * If `sources` is provided, this image will be loaded by search engines and lazy-loaded for
+     * users on browsers that don't support responsive images. If `sources` is not provided, this
+     * image will be lazy-loaded unless `disableLazyLoading` is `true`.
+     */
+    src: PropTypes.string.isRequired,
+    /**
+     * Allows the browser to choose the best file format and image size based on the device screen
+     * density and the width of the rendered image.
+     */
+    sources: PropTypes.arrayOf(
+        PropTypes.shape({
+            type: PropTypes.oneOf(['image/webp', 'image/jpeg', 'image/png', 'image/gif'])
+                .isRequired,
+            srcSet: PropTypes.string.isRequired,
+        }),
+    ),
+    alt: PropTypes.string,
+    /**
+     * Crops the image at the provided height. The `objectFit` and `objectPosition` props can be
+     * used to control how the image is cropped.
+     */
+    height: PropTypes.string,
+    /**
+     * Crops the image to a specific aspect ratio and creates a [placeholder box](https://css-tricks.com/aspect-ratio-boxes/)
+     * for the image. The placeholder prevents the browser scroll from jumping when the image is
+     * lazy-loaded.
+     */
+    aspectRatio: PropTypes.string,
```

Comment:
IMHO, this prop should be a decimal representation of the ratio rather than a string.  While there's a readability aspect to a string, it's also more limiting and introduces a format concern that isn't really relevant to this component.  Any consumer should be capable of providing the decimal version, and in some situations might end up having to convert from another format to colon separated, and back.

Reply:
I initially had it as a decimal, but changed due to feedback from @tomgenoni. Agree that the decimal is simpler, but I can see it being confusing since `ImageServiceProvider` uses strings. I'm not sure which approach I prefer.

Reply:
I would argue that ImageServiceProvider is a separate concern from this component and os it shouldn't be enforced that Image uses the same format.   I think it makes sense for ImageServiceProvider to take a string since it supports a limited set of ratios, but the either 
1) the aspect ratio is passed to render props as a decimal or

Reply:
Yeah, I think that's a fair point. @tomgenoni — what do you think?

Reply:
I understand the point but my concern was that we have same named props in close proximity that require different values for reasons that are complicated to explain (and no one will remember). My preference is that we don't confuse an already complicated set of components in this way.

Could it be a fraction `8/5`? The decimal requires determining the width/height of the image, doing math and then how many decimal places should I use...? Seems like extra work.

Reply:
> My preference is that we don't confuse an already complicated set of components in this way.

Completely agree. do you think renaming the passthrough prop on ImageServiceProvider to `placeholderRatio` (a decimal value) would solve this?  Ideally you would not need to remap any props for most use cases:

```
<ImageServiceProvider id="123456" aspectRatio="16:9">
    // imageProps is an object that includes { placeholderRatio: 1.7777 }
    {(imageProps) => <Image {...imageProps} />}
</ImageServiceProvider>
```

Reply:
In this case `<ImageServiceProvider>` does the logic of converting from 16:9 to 1.7777. `<Image>` will only check `0 < placeholderRatio < 2`.